x-pack/plugins/stack_alerts/common/build_sorted_events_query.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ESSearchBody, ESSearchRequest } from '../../../typings/elasticsearch';
+import { SortOrder } from '../../../typings/elasticsearch/aggregations';
+
+type BuildSortedEventsQueryOpts = Pick<ESSearchBody, 'aggs'> &
+  Pick<Required<ESSearchRequest>, 'index' | 'size'>;
+
+export interface BuildSortedEventsQuery extends BuildSortedEventsQueryOpts {
+  filter: unknown;
+  from: string;
+  to: string;
+  sortOrder?: SortOrder | undefined;
+  searchAfterSortId: string | number | undefined;
+  timeField: string;
+}
+
+export const buildSortedEventsQuery = ({
+  aggs,
+  index,
+  from,
+  to,
+  filter,
+  size,
+  searchAfterSortId,
+  sortOrder,
+  timeField,
+}: BuildSortedEventsQuery): ESSearchRequest => {
+  const sortField = timeField;
+  const docFields = [timeField].map((tstamp) => ({
+    field: tstamp,
+    format: 'strict_date_optional_time',
+  }));
+
+  const rangeFilter: unknown[] = [
+    {
+      range: {
+        [timeField]: {
+          lte: to,
+          gte: from,
+          format: 'strict_date_optional_time',
+        },
+      },
+    },
+  ];
+  const filterWithTime = [filter, { bool: { filter: rangeFilter } }];
+
+  const searchQuery = {
+    allowNoIndices: true,
+    index,
+    size,
+    ignoreUnavailable: true,
+    track_total_hits: true,
```

Comment:
Is it worth making `track_total_hits ` configurable? 🤔 
In a large data set setting this to true will impact performance, and I can see situations where I care more about knowing that something is happening than knowing all the instances in which it happens.

I haven't thought about this in depth, so I'm likely missing something, but I'm not 100% sure why we need to specify this.

Reply:
I can make it configurable, but I think setting it to `true` for a threshold alert would make the most sense since we want to compare the actual count to the threshold condition and setting `track_total_hits` to false would cap the count at 10,000. Does that make sense?

Reply:
Yeah, makes sense 👍

Reply:
Updated to be configurable.